rosco-web/src/main/groovy/com/netflix/spinnaker/rosco/controllers/V2BakeryController.java

```diff
@@ -1,24 +1,31 @@
 package com.netflix.spinnaker.rosco.controllers;
 
 import com.netflix.spinnaker.kork.artifacts.model.Artifact;
-import com.netflix.spinnaker.rosco.manifests.BakeManifestRequest;
-import com.netflix.spinnaker.rosco.manifests.helm.HelmBakeManifestRequest;
-import com.netflix.spinnaker.rosco.manifests.helm.HelmBakeManifestService;
+import com.netflix.spinnaker.rosco.manifests.BakeManifestService;
 import groovy.util.logging.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Map;
 
 @RestController
 @Slf4j
 public class V2BakeryController {
   @Autowired
-  HelmBakeManifestService helmBakeManifestService;
+  List<BakeManifestService> bakeManifestServices;
+
+  @RequestMapping(value = "/api/v2/manifest/bake/{type}", method = RequestMethod.POST)
+  Artifact doBake(@PathVariable("type") String type,  @RequestBody Map<String, Object> request) {
```

Comment:
What's responsible for converting the request body to the right concrete class? Is it the `BakeManifestService` class?

Reply:
**KustomizeBakeManifestService** and **HelmBakeManifestService** implements **BakeManifestService**, both have the method “handles” which in conjunction with the “type” in the path determine which service the request should be treated and then in the Service is converted: 
```
public Artifact bake(Map<String, Object> request) {
  ObjectMapper mapper = new ObjectMapper();
  KustomizeBakeManifestRequest bakeManifestRequest = mapper.convertValue(request, KustomizeBakeManifestRequest.class);
```